src/util/parallel/threadpool.no-domainslib.ml

```diff
@@ -0,0 +1,17 @@
+module Thread_pool =
+struct
+
+  let run pool f = f ()
+
+  let create n = ()
+
+  let add_work pool f =
+    let promise = Domain.spawn f in
```

Comment:
The dummy implementation is supposed to be OCaml 4 compatible, so it can't use `Domain` at all.

Reply:
Domain is provided by `domain_shims`, which is already included here. It acts as `Domain` on Ocaml5 and provides async parallelism on OCaml4.

Reply:
domain_shims implements `Domain` via `Thread` though, so it still allows interleaved execution (but not parallel execution). Even for that, output needs to be done under a real mutex, not a dummy one.
If the non-parallel mutex is a dummy, then the non-parallel thread pool probably should be even stupider: just run in the same thread entirely.

Reply:
The disadvantage of using a real mutex in the Dummy would be, that we introduce that overhead in the single threaded case wherever we use the mutexes.

Since we do not plan to use this absence of true parallelism, running in the same thread entirely seems to be the better option. (We could also consider just throwing an error, but running in the same thread is OK IMO)